File: java/src/main/java/org/urlshortener/core/shorturl/ShortUrlService.java
package org.urlshortener.core.shorturl;

import lombok.NonNull;
import org.urlshortener.core.dtos.CreateShortUrlDto;
import org.urlshortener.core.dtos.ShortUrlDto;
import org.urlshortener.core.dtos.UpdateShortUrlDto;

import java.util.List;

public interface ShortUrlService {
    ShortUrlDto findByShortname(@NonNull String shortName);

    List<ShortUrlDto> findAll(long userId);

    ShortUrlDto findById(long id, long userId);

    ShortUrlDto update(@NonNull UpdateShortUrlDto updateShortUrlDto);

    void delete(long id, long userId);

    ShortUrlDto create(@NonNull CreateShortUrlDto createShortUrlDto);
}

File: src/main/kotlin/com/urlshortener/core/services/shorturl/ShortUrlService.kt
package com.urlshortener.core.services.shorturl

import com.urlshortener.core.dtos.CreateShortUrlDto
import com.urlshortener.core.dtos.ShortUrlDto
import com.urlshortener.core.dtos.UpdateShortUrlDto
import org.springframework.stereotype.Service

@Service
interface ShortUrlService {
    fun findByShortname(shortName: String): ShortUrlDto
    fun findAll(userId: String): List<ShortUrlDto>
    fun findById(id: Long, userId: String): ShortUrlDto
    fun update(entity: UpdateShortUrlDto): ShortUrlDto
    fun delete(id: Long, userId: String)
    fun create(createShortUrlDto: CreateShortUrlDto): ShortUrlDto
}